main.py

```diff
@@ -97,6 +98,41 @@ def get_plugin_link_mapping(config, access_token):
     data_sets = list_response.json()['BrightspaceDataSets']
     return { d['PluginId']: d['DownloadLink'] for d in data_sets }
 
+def get_number_of_columns(db_conn_params, table):
+    with psycopg2.connect(**db_conn_params) as conn:
+        with conn.cursor() as cur:
+            cur.execute(
+                sql.SQL('''
+                    SELECT *
+                    FROM {table}
+                    LIMIT 0;
+                ''')
+                .format(
+                    table=sql.Identifier(table)
+                )
+            )
+
+            return len(cur.description)
+
+def process_csv_stream(csv_input_stream, num_columns_in_table):
+    '''
+    Ignore excessive columns in the CSV due to additive changes / BDS minor
+    changes by ignoring any columns in the CSV past the number of columns in the
+    table
+    '''
+    csv_rows = []
+    csv_reader = csv.reader(csv_input_stream, quoting=csv.QUOTE_MINIMAL)
+    for line in csv_reader:
+        csv_rows.append(line[:num_columns_in_table])
+
+    csv_data = io.StringIO()
+    csv_writer = csv.writer(csv_data, quoting=csv.QUOTE_MINIMAL)
+    csv_writer.writerows(csv_rows)
+
+    # Rewind the stream to the beginning before returning it
+    csv_data.seek(io.SEEK_SET)
```

Comment:
Potentially unnecessary as we don't guarantee sort order in the CSV file - probably not a big deal just thought you should know :)